object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/CalledMethodsPredicateEvaluator.java

```diff
@@ -15,6 +29,118 @@ public CalledMethodsPredicateEvaluator(final Set<String> cmMethods) {
     this.cmMethods = cmMethods;
   }
 
+  /**
+   * Return true iff the boolean formula "lhs -> rhs" is valid, treating all variables as
+   * uninterpreted.
+   *
+   * <p>Used to determine subtyping between two @CalledMethodsPredicate annotations that are not
+   * identical.
+   */
+  public static boolean implies(final String lhsOrig, final String rhsOrig) {
+
+    JavaParser parser = new JavaParser();
+
+    String preamble = "class DUMMY { boolean DUMMY() { return ";
```

Comment:
It's a bit weird that the flow of code in this method is:
 * parsing-related
 * set up solver
 * more parsing-related

Please move all the parsing-related code together.  Once doing that, it becomes clear that lines 43-54 should be merged into `compilationUnitToBooleanFormula` to avoid duplication.  (That method might need to be slightly renamed.)